### PR TITLE
Remove Crypto from pruning crypto

### PR DESCRIPTION
**Pull request by yontyon**

### CODE REVIEW

File: kvbc/include/pruning_handler.hpp [hunk 1]
```diff
@@ -75,7 +72,7 @@ class RSAPruningVerifier {
  private:
   struct Replica {
     std::uint64_t principal_id{0};
-    bftEngine::impl::RSAVerifier verifier;
+    std::unique_ptr<concord::util::crypto::IVerifier> verifier;
```
Comment (toly-kournik) [inline kvbc/include/pruning_handler.hpp:75]: If a class is `RSAPrunningVerifier`, shouldn't there be a pointer to `concord::util::crypto::RSAVerifier`?

File: kvbc/include/pruning_handler.hpp [hunk 2]
```diff
@@ -40,13 +40,10 @@ class RSAPruningSigner {
...
   // signature.
-  void sign(concord::messages::LatestPrunableBlock &) const;
+  void sign(concord::messages::LatestPrunableBlock &);
 
  private:
-  std::string getSignatureBuffer() const;
-
- private:
-  bftEngine::impl::RSASigner signer_;
+  std::unique_ptr<concord::util::crypto::ISigner> signer_;
```
Comment (toly-kournik) [inline kvbc/include/pruning_handler.hpp:46]: Same as Verifier

File: kvbc/include/pruning_handler.hpp (hunk 1 above)
Comment (yontyon) [reply kvbc/include/pruning_handler.hpp:75]: We don't.
I didn't want to have too much changes, so I left it to a future refactoring

**Review by toly-kournik — APPROVED** [review]
LGTM!

**Pull request merged**